include/sound/sof/ipc4/header.h

```diff
@@ -108,6 +108,11 @@ enum sof_ipc4_global_msg {
 
 	/* Loads library (using Code Load or HD/A Host Output DMA) */
 	SOF_IPC4_GLB_LOAD_LIBRARY,
+	/*
+	 * Prepare the host DMA channel for library loading, must be followed by
+	 * a  SOF_IPC4_GLB_LOAD_LIBRARY message as the library loading step
+	 */
+	SOF_IPC4_GLB_LOAD_LIBRARY_PREPARE,
```

Comment:
so this is the value 25, but the comment below says it's reserved. which is it?

Reply:
what are the rules for adding new messages here? Is this not an ABI change, or is this considered to be ok because firmware has not been released just yet? The commit message MUST be clear on this.

Reply:
> so this is the value 25, but the comment below says it's reserved. which is it?

It was an unused value and the firmware will be using it for this message. Dropping the comment

Reply:
> what are the rules for adding new messages here? Is this not an ABI change, or is this considered to be ok because firmware has not been released just yet? The commit message MUST be clear on this.

As we did not have official release of IPC4 it is not yet an ABI change. We have discussed  and aligned on this internally with all teams involved with IPC4 (including the reference firmware).

`25` was reserved and not used, so firmware which does not implement this new message will reject it as unsupported.

Reply:
I think you should really add this information in the commit message. It's very helpful to put the rules in writing for all reviewers to see, including upstream.
You added information along those lines in the next patch, but that's an Intel-specific part. We should have the information on support when the header is changed.